### PR TITLE
Add reference plane option for where the vertices of polyslab is defined

### PR DESCRIPTION
**Pull request by weiliangjin2021**

### CODE REVIEW

File: tidy3d/components/geometry.py [hunk 1]
```diff
@@ -1562,13 +1562,22 @@ class PolySlab(Planar):
         units=RADIAN,
     )
 
+    reference_plane: PlanePosition = pydantic.Field(
+        "bottom",
+        title="Reference plane of polygon vertices",
+        description="The position of the plane where the supplied ``vertices`` are "
+        "defined. The ``vertices`` of the polygon "
+        "can be defined at the ``bottom``, ``middle``, or ``top`` of the PolySlab.",
```
Comment (tylerflex) [inline tidy3d/components/geometry.py:1570]: is `bottom` defined as the negative side of the axis? for example if axis=1 then bottom is y- side and top is y+ side? Maybe we can explicitly mention this.

**Review by tylerflex — APPROVED** [review]
Looks good, just a minor comment about how the reference plane is defined. I'm afraid I can't follow all of the changes to the polyslab vertices because of the complexity of the problem. but I trust that all of the tests are sufficient that it they would catch any known bugs if they were to arise?

File: tidy3d/components/geometry.py (hunk 1 above)
Comment (weiliangjin2021) [reply tidy3d/components/geometry.py]: yeah, that's good to add. Please checkout the new commit.

Comment (tylerflex) [reply tidy3d/components/geometry.py]: Looks good. Feel free to merge

Comment (weiliangjin2021) [reply tidy3d/components/geometry.py]: That is clearer. Edited.

**Pull request merged**